### ISSUE TITLE
question: a separate CTA for highlighting checked commits

### ISSUE THREAD

**Issue by tusharnankani** [HarshKapadia2/git-graph#19]
So initially, all the commits are highlighted and every commit is checked. 

![image](https://user-images.githubusercontent.com/61280281/162819779-03325901-11cd-41f4-9eaf-827263b5b9e8.png)

Why do we have a separate CTA? Why didn't you do it real time? Like render-on-check? 
What was your thought process?

There was this small talk by Aditya Behere ("beherebaba") about friction in products. This is adding friction.

Don't you think the UX would be better if it was rendered directly after checking the box?
Maybe it will get junky when I try to check/uncheck multiple boxes all at once, might rise to that problem, then I can understand.

![image](https://user-images.githubusercontent.com/61280281/162819727-590c9aac-3e9f-4265-9db8-f201cfd6b6d4.png)

**Comment by tusharnankani** [comment]
Now, the UX part of the buttons.

The first 2 buttons are not CTAs. They are like filters, they act like filters. 

The button at the bottom is the CTA. So it should be placed at the top maybe, because not going to lie: I actually stood there for quite a few seconds figuring out what to click for applying my checkboxes' highlights.

**Comment by HarshKapadia2** [comment]
> Maybe it will get junky when I try to check/uncheck multiple boxes all at once, might rise to that problem, then I can understand.

Yup, this is exactly the issue. It will be too janky, as renders will be caused on every selection. One possible solution is Debouncing. I have no idea if that will work here, but I should try it. Good suggestion! I'll update this issue when I try it out.

> So it should be placed at the top maybe, because not going to lie: I actually stood there for quite a few seconds figuring out what to click for applying my checkboxes' highlights.

Agreed, so I made the change in commit 8f88633b7a583ba947cba843e16ccf170b342861.

**Comment by HarshKapadia2** [comment]
> > Maybe it will get junky when I try to check/uncheck multiple boxes all at once, might rise to that problem, then I can understand.
>
> Yup, this is exactly the issue. It will be too janky, as renders will be caused on every selection. One possible solution is Debouncing. I have no idea if that will work here, but I should try it. Good suggestion! I'll update this issue when I try it out.

I implemented Debouncing to prevent the jank and removed the button for selecting checked commits. Thank you for pointing out the issue, it made a lot of sense.

**Comment by HarshKapadia2** [comment]
Could you please let me know your thoughts on the Debounce time, @tusharnankani? Should I reduce it? (Remember that reducing it too much will increase jank quite a bit.)

**Comment by tusharnankani** [comment]
Looks really, really good.
The timing is spot on. Great work! 👏🏻

The complete application looks so much better <3
